### PR TITLE
Handle WDL function sub()

### LINKED ISSUE #61: WDL function sub() is not handled
Currently, the WDL function sub() is not translated by the convertor. (https://github.com/openwdl/wdl/blob/main/versions/1.1/SPEC.md#string-substring-string-string)

A WDL workflow containing sub() function can be found here. (https://github.com/biowdl/tasks/blob/develop/bowtie.wdl#L63)

### PR DESCRIPTION
**Pull request by dinithiW**
Fixes #61

### CODE REVIEW

File: wdl2cwl/tests/cwl_files/bowtie.cwl [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            --seedlen $(inputs.seedlen) \
+            -k $(inputs.k) \
+            $(inputs["best"] ? "--best" : "") \
+            $(inputs["strata"] ? "--strata" : "") \
+            $(inputs["allowContain"] ? "--allow-contain" : "") \
+            --threads $(inputs.threads) \
+            --sam-RG '$(inputs.samRG) \
+            $(return inputs["sub(indexFiles[0]"].replace("(\.rev)?\.[0-9]\.ebwt$","");) \
+            $(inputs.sep=","readsUpstream) \
+            $(inputs.sep=","readsDownstream) \
```
Comment (mr-c) [inline wdl2cwl/tests/cwl_files/bowtie.cwl:73]: I don't think this works?

File: wdl2cwl/tests/cwl_files/bowtie.cwl [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            --sam \
+            --seedmms $(inputs.seedmms) \
+            --seedlen $(inputs.seedlen) \
+            -k $(inputs.k) \
+            $(inputs["best"] ? "--best" : "") \
+            $(inputs["strata"] ? "--strata" : "") \
+            $(inputs["allowContain"] ? "--allow-contain" : "") \
+            --threads $(inputs.threads) \
+            --sam-RG '$(inputs.samRG) \
+            $(return inputs["sub(indexFiles[0]"].replace("(\.rev)?\.[0-9]\.ebwt$","");) \
```
Comment (mr-c) [inline wdl2cwl/tests/cwl_files/bowtie.cwl:71]: Should `sub()` still be here?

**Comment by dinithiW** [comment]
Modified code as requested

File: wdl2cwl/main.py [hunk 3]
```diff
@@ -145,6 +145,35 @@ def get_command(
...
+                append_str_sub = ""
+                if search_index:
+                    append_str_sub = (
+                        '$(return inputs["'
+                        + re.sub(r"\[[0-9]\]", "", temp[0].replace("sub(", ""))
+                        + '"]'
+                        + temp[0][temp[0].find("[") :]
+                    )
+                else:
+                    append_str_sub = (
```
Comment (mr-c) [inline wdl2cwl/main.py:161]: Great! Can we get a test case with an array of files that doesn't use `[0]`?

**Review by mr-c — CHANGES_REQUESTED** [review]

**Review by mr-c — APPROVED** [review]

**Pull request merged**